### PR TITLE
Examples: Add a Hello World example using Hummingbird

### PR DESCRIPTION
**Pull request by euanh**
### Motivation

Having simple "getting started" examples is helpful for new users.   Some users may be more familiar with Hummingbird instead of Vapor.

### Modifications

Adds another example, using Hummingbird, and links it into the documentation.

### Result

New users who want to use Hummingbird can start from a tested starting point.

### Test Plan

Built and deployed the example locally, and ran the soundness checks.

### CODE REVIEW

**Comment by euanh** [comment]
Hummingbird currently fails to build with the Static Linux SDK:
* https://github.com/hummingbird-project/hummingbird/issues/571
* https://github.com/hummingbird-project/hummingbird/issues/574

File: Examples/HelloWorldVapor/Sources/main.swift [hunk 1]
```diff
@@ -21,6 +21,6 @@ let app = try Application(.detect())
 app.http.server.configuration.hostname = "0.0.0.0"
 defer { app.shutdown() }
 
-app.get { _ in "Hello World, from \(myos)\n" }
+app.get { _ in "Hello World, from Vapor on \(myos)\n" }
```
Comment (thoven87) [inline Examples/HelloWorldVapor/Sources/main.swift:24]: Should the text updated to say Hello World, from Hummingbird instead of Vapor?

Comment (euanh) [reply Examples/HelloWorldVapor/Sources/main.swift:24]: Thank you for your comment!   

In this branch we're adding a new Hummingbird example, but keeping the existing Vapor example.   This line adds `Vapor` to the Vapor example in `Examples/HelloWorldVapor/Sources/main.swift`, to make it easy to identify which example you're running.   The Hummingbird example is in a separate file at `Examples/HelloWorldHummingbird/Sources/main.swift`.

**Comment by euanh** [comment]
The Hummingbird fix has been merged but is not yet in a release.

**Comment by adam-fowler** [comment]
v2.1.0 was released yesterday.

**Comment by euanh** [comment]
> v2.1.0 was released yesterday.

🎉 

> EDIT: we might find this crashes now as [apple/swift-async-algorithms#325](https://github.com/apple/swift-async-algorithms/pull/325) is still not in a release of AsyncAlgorithms.

This simple hello world example seems to work (on x86), but maybe more complex examples will break.   I see that the issue on the hummingbird repo is on ARM - maybe that's significant?

**Comment by euanh** [comment]
> EDIT: we might find this crashes now as https://github.com/apple/swift-async-algorithms/pull/325 is still not in a release of AsyncAlgorithms

https://github.com/apple/swift-async-algorithms/releases/tag/1.0.2 has just been tagged and should have the fix in it.

**Comment by adam-fowler** [comment]
Oh awesome!

**Comment by euanh** [comment]
The example server did eventually crash after sitting idle for a while.   This build was using the previous release of async-algorithms, and seemed to work initially.   I'll try it with the new release.

```
2024-10-08T16:06:30+0000 info Hummingbird : [HummingbirdCore] Server started and listening on 0.0.0.0:8080
AsyncAlgorithms/MergeStateMachine.swift:589: Fatal error: Invalid state
Current stack trace:
0    <unknown>                          0x000000000265d281
1    <unknown>                          0x0000000002256277
2    <unknown>                          0x0000000002256083
3    <unknown>                          0x00000000022564d9
4    <unknown>                          0x00000000024152e6
5    <unknown>                          0x000000000179b487
6    <unknown>                          0x000000000179c6fb
7    <unknown>                          0x00000000026a2b23
8    <unknown>                          0x00000000026a3a95
9    <unknown>                          0x000000000275a5cc
10   <unknown>                          0x000000000275a439
11   <unknown>                          0x0000000002763b46
```

**Comment by adam-fowler** [comment]
Others have verified that AsyncAlgorithms v1.0.2 fixes this issue.

**Pull request merged**